### PR TITLE
wlclock: init at 1.0.1

### PR DESCRIPTION
**Pull request by NomisIV**
###### Description of changes

Packaged wlclock (https://github.com/NixOS/nixpkgs/issues/157801)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/wlclock/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+, cmake
+, pkg-config
+, wayland-protocols
+, wayland
+, cairo
+, scdoc
+}:
+stdenv.mkDerivation rec {
+  pname = "wlclock";
+  version = "v1.0.1";
```
Comment (legendofmiracles) [inline pkgs/applications/misc/wlclock/default.nix:16]:
```suggestion
  version = "1.0.1";
```

File: pkgs/applications/misc/wlclock/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  src = fetchFromSourcehut {
+    owner = "~leon_plickat";
+    repo = "wlclock";
+    rev = version;
```
Comment (legendofmiracles) [inline pkgs/applications/misc/wlclock/default.nix:21]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/misc/wlclock/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  buildInputs = with pkgs; [
+    wayland-protocols
+    wayland
+    cairo
+    scdoc
+  ];
+
+  meta = with lib; {
+    description = "a digital analog clock for Wayland desktops.";
```
Comment (legendofmiracles) [inline pkgs/applications/misc/wlclock/default.nix:40]:
```suggestion
    description = "A digital analog clock for Wayland desktops";
```

File: pkgs/applications/misc/wlclock/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{ pkgs
+, stdenv
+, lib
```
Comment (legendofmiracles) [inline pkgs/applications/misc/wlclock/default.nix:3]:
```suggestion
{ lib
, stdenv
```
`pkgs` isn't used and lib should always be first

File: pkgs/applications/misc/wlclock/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+, fetchFromSourcehut
+, meson
+, ninja
+, cmake
+, pkg-config
+, wayland-protocols
+, wayland
+, cairo
+, scdoc
+}:
```
Comment (legendofmiracles) [inline pkgs/applications/misc/wlclock/default.nix:13]:
```suggestion
}:

```
nitpick

File: pkgs/applications/misc/wlclock/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+  buildInputs = with pkgs; [
+    wayland-protocols
+    wayland
+    cairo
+    scdoc
+  ];
+
+  meta = with lib; {
+    description = "A digital analog clock for Wayland desktops.";
```
Comment (legendofmiracles) [inline pkgs/applications/misc/wlclock/default.nix:40]:
```suggestion
    description = "A digital analog clock for Wayland desktops";
```
another nitpick, sorry

**Review by legendofmiracles — APPROVED** [review]
Other than these small things LGTM

Can't test though, maybe someone else can merge

File: pkgs/applications/misc/wlclock/default.nix (hunk 4 above)
Comment (NomisIV) [reply pkgs/applications/misc/wlclock/default.nix]: Thanks! I totally missed that xD

**Pull request merged**